packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -58,20 +62,31 @@ class PopupMenuItem<T> extends PopupMenuEntry<T> {
 
   double get height => _kMenuItemHeight;
 
+  _PopupMenuItemState<PopupMenuItem<T>> createState() => new _PopupMenuItemState<PopupMenuItem<T>>();
+}
+
+class _PopupMenuItemState<T extends PopupMenuItem> extends State<T> {
+  // Override this to put something else in the menu entry.
+  Widget buildChild() => config.child;
+
+  void onTap(BuildContext context) {
```

Comment:
I think you can just use the State's context property here; don't need to create a closure around onTap below.